### PR TITLE
Fetch fewer repo edges from ZenHub to improve performance

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: helpers/repos.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    Repo(('zcash', 'librustzcash'), 85334928, 'Z2lkOi8vcmFwdG9yL1JlcG9zaXRvcnkvMTg5MDU1NTE'),
+    Repo(('zcash', 'lightwalletd'), 159714694, 'Z2lkOi8vcmFwdG9yL1JlcG9zaXRvcnkvMTg5MDU1NzE'),
+] + ANDROID_REPOS + IOS_REPOS
+
+ECC_REPOS = CORE_REPOS + TFL_REPOS + WALLET_REPOS + [
+    Repo(('Electric-Coin-Company', 'infrastructure'), 65419597, 'Z2lkOi8vcmFwdG9yL1JlcG9zaXRvcnkvMjIwNTA1NjY'),
+]
+
+ZF_REPOS = [
+    Repo(('ZcashFoundation', 'zebra'), 205255683, None),
```
Comment (str4d) [inline helpers/repos.py:72]: Turns out that the ZenHub GraphQL API key doesn't actually let us enumerate repos, so we'd need ZF to run the new script I added to get those ZenHub IDs.

File: helpers/zenhub.py [hunk 2]
```diff
@@ -24,11 +33,36 @@ def api(token):
...
+
+    return repos
+
+
 def fetch_workspace_graph(op, workspace_id, repos, cursor):
     dependencies = op.workspace(id=workspace_id).issue_dependencies(
-        # TODO: This causes a 500 Internal Server Error. We need the ZenHub repo IDs here,
-        # not the GitHub repo IDs (which the previous REST API used).
-        # repository_ids=repos,
+        repository_ids=[repo.zh_id for repo in repos],
```
Comment (str4d) [inline helpers/zenhub.py:65]: I wouldn't have needed to do any of this if ZenHub supported `repositoryGhIds` here like it does elsewhere. But `repositoryIds` takes ZenHub IDs.

File: helpers/repos.py [hunk 3]
```diff
@@ -1,75 +1,90 @@
...
+        self.name = name
+        self.gh_id = gh_id
+
+    def __repr__(self):
+        if self.name:
+            repo = self.name
+            # Shorten the representation of long repo names.
+            if repo[0] == 'Electric-Coin-Company':
+                repo = ('ECC', repo[1])
+            return '/'.join(repo)
```
Comment (nuttycom) [inline helpers/repos.py:13]: `name` being a tuple (or a list?) and then referred to in the `Repo` class as `repo` is very confusing. Why not have three properties: `org`, `repo`, and `gh_id`?

**Review by nuttycom — APPROVED** [review]
utACK with one minor suggestion for improvement (can be deferred until later, or added on)

File: helpers/repos.py (hunk 3 above)
Comment (str4d) [reply helpers/repos.py:13]: `org` and `repo` are collectively either present or absent; you don't get one without the other in the places we have it. So it makes sense to be in effect `Option<(&str, &str)>`. `repo` just happens to be what the original code named it.

**Pull request merged**